tests/unit/test_scheduler.py

```diff
@@ -1,10 +1,63 @@
 import mock
+import getpass
 import pytest
 
-from scc_hypervisor_collector.api import exceptions, CredentialsConfig, CollectorConfig, CollectionScheduler, SccCredsConfig, HypervisorCollector
+from scc_hypervisor_collector.api import (
+    exceptions, CredentialsConfig, CollectorConfig,
+    CollectionResults, CollectionScheduler,
+    SccCredsConfig, HypervisorCollector
+)
 from tests import utils
 
 
+class TestCollectionResults:
+
+
+    @pytest.mark.config('tests/unit/data/collected/libvirt/collector.results')
+    def test_collection_results_contents(self, collected_results):
+        libvirt1 = collected_results.results[0]
+        assert libvirt1['backend'] == 'libvirt1'
+        assert libvirt1['valid'] == True
+        libvirt1_details = libvirt1['details']['virtualization_hosts'][0]
+        assert libvirt1_details['group_name'] == 'libvirt1'
+        assert libvirt1_details['properties']['name'] == 'libvirt1.example.com'
+        assert libvirt1_details['systems'][0]['properties']['vm_name'] == 'test-sle12'
+
+    @pytest.mark.config('tests/unit/data/collected/libvirt/collector.results')
+    def test_collection_results_load(self, collected_results, tmp_path):
+        results_file = tmp_path / 'collected.results'
+        results_file.touch(mode=0o600)  # ensure results file exists with correct mode
+        with mock.patch('yaml.safe_load',
+                        return_value=collected_results.results) as yaml_safe_load:
+            collected_results.load(results_file)
+            yaml_safe_load.assert_called_once()
+
+    @pytest.mark.config('tests/unit/data/collected/libvirt/collector.results')
+    def test_collection_results_save(self, collected_results, tmp_path):
+        results_file = tmp_path / 'collected.results'
+        with mock.patch('yaml.safe_dump') as yaml_safe_dump:
+            collected_results.save(results_file)
+            yaml_safe_dump.assert_called_once()
+
+    @pytest.mark.config('tests/unit/data/collected/libvirt/collector.results')
+    def test_collection_results_load_invalid_perms(self, collected_results, tmp_path):
+        results_file = tmp_path / 'collected.results'
+        results_file.touch(mode=0o640)  # ensure results file exists with correct mode
```

Comment:
In this case I'm explicitly changing the permissions on the file so that it will trigger the failure case, because the file permission is not 0o600. So this one I want to keep in, but I need to fix the comment to match the code. ;-)